bip_converter/t3dn_bip_converter/convert.py

```diff
@@ -24,38 +25,48 @@ def convert_file(src: Union[str, Path], dst: Union[str, Path] = None):
         raise ValueError('exactly one file must be in BIP format')
 
 
-def _image_to_bip(src: Path, dst: Path):
+def _image_to_bip(src: Union[str, Path], dst: Union[str, Path]):
     '''Convert various image formats to BIP.'''
     with Image.open(src) as image:
         image = image.transpose(Image.FLIP_TOP_BOTTOM)
-        image = image.convert('RGBa')
 
-        width, height = image.size
-        data = image.tobytes()
+        if not image.mode.endswith(('A', 'a')):
+            image = image.convert('RGBA')
+        elif image.mode != 'RGBa':
+            image = image.convert('RGBa')
+
+        images = [image.resize(size=(32, 32)), image]
+        contents = [compress(image.tobytes()) for image in images]
 
         with open(dst, 'wb') as output:
-            output.write(b'BIP1')
+            output.write(b'BIP2')
+            output.write(len(images).to_bytes(1, 'big'))
 
-            output.write(width.to_bytes(2, 'big'))
-            output.write(height.to_bytes(2, 'big'))
+            for image, content in zip(images, contents):
+                for number in image.size:
+                    output.write(number.to_bytes(2, 'big'))
+                output.write(len(content).to_bytes(4, 'big'))
 
-            output.write(compress(data))
+            for content in contents:
+                output.write(content)
 
 
-def _bip_to_image(src: Path, dst: Path):
+def _bip_to_image(src: Union[str, Path], dst: Union[str, Path]):
     '''Convert BIP to various image formats.'''
     with open(src, 'rb') as bip:
-        magic = bip.read(4)
-
-        if magic != b'BIP1':
+        if bip.read(4) != b'BIP2':
             raise ValueError('input is not a supported file format')
 
-        width = int.from_bytes(bip.read(2), 'big')
-        height = int.from_bytes(bip.read(2), 'big')
+        count = int.from_bytes(bip.read(1), 'big')
```

Comment:
I think you should add an assert, that the `count` is at least `1`. I think the following code works fine if `count` is larger than `0`. Is my interpretation correct?